campus/fhda/fhda_scrape.py

```diff
@@ -151,3 +109,73 @@ def transform_class(class_data):
 
     with open(join(DB_DIR, 'metadata.json'), 'w') as outfile:
         json.dump({'tags': dict(tagdbs), 'terms': dict(termdbs)}, outfile)
+
+    log_info('Dumped metadata')
+
+
+def run_advanced_scraper(**kwargs):
+    try:
+        scraper = AdvancedScraper(
+            ssb_url=SSB_URL,
+            db_dir=DB_DIR,
+            cache_dir=CACHE_DIR,
+            hooks=FHDAScraperHooks,
+            login=login,
+
+            trace=True,
+            **kwargs,
+        )
+        scraper.run()
+
+    except KeyboardInterrupt:
+        log_err('Aborted advanced schedule scraper', start='\n')
+
+
+def run_public_schedule_scraper(**kwargs):
+    try:
+        scraper = ScheduleScraper(
+            ssb_url=SSB_URL,
+            db_dir=DB_DIR,
+            cache_dir=CACHE_DIR,
+            hooks=FHDAScraperHooks,
+            # login=login,
+
+            trace=True,
+            **kwargs,
+        )
+        scraper.run()
+
+    except KeyboardInterrupt:
+        log_err('Aborted public schedule scraper', start='\n')
+
+
+if __name__ == '__main__':
+    ENABLE_ADVANCED = True
+    ENABLE_SCHEDULE = True
+
+    parser = argparse.ArgumentParser(description='Scrape FHDA data')
+    parser.add_argument('--update', default=False, action='store_true',
+                        help='Update data with seat info')
+    args = parser.parse_args()
+
+    if args.update:
+        run_public_schedule_scraper(
+            max_terms=2,
+            use_cache=False
+        )
+        write_metadata()
+        scrape_seats()
+    else:
+        if ENABLE_ADVANCED:
+            run_advanced_scraper(
+                max_terms=8,
+                # use_cache=False,
+                start_term='202112',
+            )
```

Comment:
TODO: parameters here will be customizable through CLI options when the scraper refactor is done